### PR TITLE
Removed all references to the THREE namespace

### PR DESCRIPTION
**Pull request by Benjamin-Dobell**
The first step towards making three.js `jsnext:main` compatible (https://github.com/mrdoob/three.js/pull/9402).

I've actually implemented this _without_ any string to constructor maps. The logic behind the changes is that there's only ever one source of truth for these objects and constants.

e.g. If you add a new (extra) type of Geometry you only need to remember to export it _one_ place (`src/extras/geometries/Geometries.js`), and it'll work project wide and Loaders will pick it up automatically.

### CODE REVIEW

**Comment by Benjamin-Dobell** [comment]
Whoops, wait a tick, seems I included some duplicate exports... fixing now...

**Comment by Benjamin-Dobell** [comment]
Fixed 😄

**Comment by GGAlanSmithee** [comment]
Nice! A question (which is related to, but not caused by, this PR) regarding `import`s; is it preffered to omit the `.js` from the filename? I Always thought it was, but there seems to be some inconsistencies in `three.js` now. I guess there is no functional diffetence though..

**Comment by Benjamin-Dobell** [comment]
@GGAlanSmithee Actually, I was wondering the same thing!

My gut feel was that `.js` extensions were somehow more portable in the entry point (`jsnext:main`) file, say if you're using a packager other than rollup. However, I have no idea why that would be the case.

If my hypothesis is somehow correct, then this pull request presently breaks the rule anyway, as I've copied and paste the exports leaving the `.js` extension intact, where I guess it shouldn't be.

@Rich-Harris Sorry to drag you in, is there any reason for including the `.js` extension in the imported `module-name`?

**Comment by Rich-Harris** [comment]
@Benjamin-Dobell no worries! Yes, there is a good reason – when browsers eventually support `import` and `export` natively, they almost certainly won't allow you to omit the extension. (We're planning to remove support for extension-less imports in Rollup to match that behaviour. When we do, we'll release a CLI tool that tries to fix projects that currently rely on them.)

Extension-less `require(...)` statements were a mistake made in the early days of Node (in the name of convenience and aesthetics, presumably) – they're needlessly ambiguous and cause extra work for everyone (it's part of the reason [require is dog slow](https://kev.inburke.com/kevin/node-require-is-dog-slow/)). Don't use them! 😀

**Comment by GGAlanSmithee** [comment]
Thanks for the explanation @Rich-Harris, I did not know.

File: src/loaders/ObjectLoader.js [hunk 1]
```diff
@@ -26,11 +28,17 @@ import { MaterialLoader } from './MaterialLoader';
...
 import { JSONLoader } from './JSONLoader';
 import { XHRLoader } from './XHRLoader';
+import * as CoreGeometry from '../core/Geometries';
+import * as ExtraGeometry from '../extras/geometries/Geometries';
 
 /**
  * @author mrdoob / http://mrdoob.com/
  */
 
+function resolveGeometryType ( type ) {
```
Comment (sirxemic) [inline src/loaders/ObjectLoader.js]: Is this function actually necessary? As far as I can see the function is only used for geometry types which are part of the `ExtraGeometry` collection.

File: src/Three.js [hunk 2]
```diff
@@ -116,10 +102,8 @@ export { AnimationMixer } from './animation/AnimationMixer.js';
 export { AnimationClip } from './animation/AnimationClip.js';
 export { AnimationAction } from './animation/AnimationAction.js';
 export { Uniform } from './core/Uniform.js';
-export { InstancedBufferGeometry } from './core/InstancedBufferGeometry.js';
-export { BufferGeometry } from './core/BufferGeometry.js';
-export { DirectGeometry } from './core/DirectGeometry.js';
-export { GeometryIdCount, Geometry } from './core/Geometry.js';
+export * from './core/Geometries.js';
```
Comment (mrdoob) [inline src/Three.js]: I like `./materials/Material.js` and `./extras/geometries/Geometries.js` but I wouldn't this `./core/Geometries.js` feels like over-engineering to me.

File: src/loaders/ObjectLoader.js (hunk 1 above)
Comment (mrdoob) [inline src/loaders/ObjectLoader.js]: Agreed! Relates to my other over-engineer comment.

Comment (Benjamin-Dobell) [inline src/loaders/ObjectLoader.js]: Haha, I wouldn't call it so much over-engineered, as I'd say general stupidity on my own behalf! :wink:

I have a bad tendency when submitting pull requests to try make things operate similarly to their prior implementation, even when it no longer makes any sense!

I'll sort it out shortly.

Comment (mrdoob) [inline src/loaders/ObjectLoader.js]: Sounds good! Leave a note when you do so I don't forget about this 😅

Comment (mrdoob) [inline src/loaders/ObjectLoader.js]: I guess I'll have to do it myself 😜

**Comment by mrdoob** [comment]
Thanks!

File: src/loaders/ObjectLoader.js (hunk 1 above)
Comment (Benjamin-Dobell) [inline src/loaders/ObjectLoader.js]: @mrdoob I'm so sorry! I've been flat-out with work recently and being in the process of buying a house has sucked up most of my free time. Thanks, for taking over, much appreciated!

**Comment by mrdoob** [comment]
@takahirox 

Seems like this PR broke `MMDLoader` because `ObjectLoader` now uses an internal list of possible materials so it never gets to access `MMDMaterial`.

Can you think of a workaround to get it working again? What does `MMDMaterial` do that `MeshPhongMaterial` doesn't?

**Comment by takahirox** [comment]
I'll check later...

**Comment by Benjamin-Dobell** [comment]
If `MMD` really does require custom material functionality, then I would guess that a custom `ShaderMaterial` would be a more appropriate implementation - mind you I'm yet to write any custom materials myself.

If we need backwards compatibility then `MMDMaterial` can be added to `Materials.js`, and it would just thin wrapper around the new `ShaderMaterial` based implementation.

**Pull request merged**